streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkClusterWatcher.java

```diff
@@ -0,0 +1,248 @@
+      watcher();
+    }
+  }
+
+  private void watcher() {
+    lastWatcheringTime = System.currentTimeMillis();
+    for (Map.Entry<Long, FlinkCluster> entry : WATCHER_CLUSTERS.entrySet()) {
+      EXECUTOR.execute(
+          () -> {
+            FlinkCluster flinkCluster = entry.getValue();
+            Integer clusterExecutionMode = flinkCluster.getExecutionMode();
+            if (!ExecutionMode.isKubernetesSessionMode(clusterExecutionMode)) {
+              ClusterState state = getClusterState(flinkCluster);
+              handleClusterState(flinkCluster, state);
+            } else {
+              // TODO: K8s Session status monitoring
+            }
+          });
+    }
+  }
+
+  /**
+   * cluster get state from flink or yarn api
+   *
+   * @param flinkCluster
+   * @return
+   */
+  private ClusterState getClusterState(FlinkCluster flinkCluster) {
+    ClusterState state = getClusterStateFromFlinkAPI(flinkCluster);
+    if (ClusterState.isRunningState(state)) {
+      return state;
+    } else {
+      return getClusterStateFromYarnAPI(flinkCluster);
+    }
+  }
+
+  /**
+   * cluster get state from flink rest api
+   *
+   * @param flinkCluster
+   * @return
+   */
+  private ClusterState getClusterStateFromFlinkAPI(FlinkCluster flinkCluster) {
+    final String address = flinkCluster.getAddress();
+    final String jobManagerUrl = flinkCluster.getJobManagerUrl();
+    if (StringUtils.isEmpty(address)) {
+      return ClusterState.STOPPED;
+    }
+    final String flinkUrl =
+        StringUtils.isEmpty(jobManagerUrl)
+            ? address.concat("/overview")
+            : jobManagerUrl.concat("/overview");
+    try {
+      String res =
+          HttpClientUtils.httpGetRequest(
+              flinkUrl, RequestConfig.custom().setConnectTimeout(5000).build());
+
+      JacksonUtils.read(res, Overview.class);
+      return ClusterState.RUNNING;
+    } catch (Exception ignored) {
+      log.error("cluster id:{} get state from flink api failed", flinkCluster.getId());
+    }
+    return ClusterState.UNKNOWN;
+  }
+
+  /**
+   * cluster get state from yarn rest api
+   *
+   * @param flinkCluster
+   * @return
+   */
+  private ClusterState getClusterStateFromYarnAPI(FlinkCluster flinkCluster) {
+    if (ExecutionMode.isRemoteMode(flinkCluster.getExecutionModeEnum())) {
+      return ClusterState.STOPPED;
+    }
+    String clusterId = flinkCluster.getClusterId();
+    if (StringUtils.isEmpty(clusterId)) {
+      return ClusterState.STOPPED;
+    }
+    String yarnUrl = "ws/v1/cluster/apps/".concat(flinkCluster.getClusterId());
+    try {
+      String result = YarnUtils.restRequest(yarnUrl);
+      if (null == result) {
+        return ClusterState.UNKNOWN;
+      }
+      YarnAppInfo yarnAppInfo = JacksonUtils.read(result, YarnAppInfo.class);
+      ClusterState state = ClusterState.of(yarnAppInfo.getApp().getFinalStatus());
+      if (ClusterState.UNDEFINED.equals(state)) {
```

Comment:
The Yarn resourcemanager rest API will include special cases where the state is accepted but the final status is UNDEFINED.

So it would be better  to judge the current Yarn session status based on `state` and `finalStatus` of yarnAPP.

Reply:
> The Yarn resourcemanager rest API will include special cases where the state is accepted but the final status is UNDEFINED.
> 
> So it would be better to judge the current Yarn session status based on `state` and `finalStatus` of yarnAPP.

Here, I will reconsider the processing logic.